demo/demo.less

```diff
@@ -32,6 +32,10 @@
 .mdl-button .material-icons {
   line-height: 36px;
 }
+//icon is not at the center. this fixes it.
+.mdl-button .material-icons-round {
+  padding-top: 3px;
```

Comment:
Is this value dependent on the font size or line height?  It would be best to find a robust fix that doesn't have a value dependent on other settings.

Reply:
I don't think it depends  upon line height. Actually  hover effect is applied on .mdl-button and there is not enough space from top for .materials-icons-round (x). hence I found  padding to solve the issue.

Reply:
This doesn't seem right.  The content of the button is not vertically aligned to the middle, and adding padding to the top is a hack.  If you reduce the font size, this isn't needed at all, so I don't think it's the right fix.

Let's experiment a bit and see if we can determine a better solution.

Reply:
I think I found a better solution.

The button is a `display: block` at 32x32 and the icon is a `display: inline-block` at 24x24.  Buttons *should* align their contents to the center by default, but something in MDL may be preventing this from working correctly.

Switching to flex layout makes this easy to fix, though:

```css
.mdl-button--icon {
  /* By default, MDL icon buttons may end up vertically off-center.  This aligns
   * them to the center both vertically and horizontally. */
  display: flex;
  align-items: center;
  justify-content: center;
}
```

What do you think?

Reply:
Yaah! it actually works fine!
gosh! why didn't I think about it!!!!

Reply:
There are so many ways to do something in CSS, and experience has taught us that some of them are brittle and cause problems further down the road.  We have a TODO in a few places to eliminate hard-coded pixel values for things, because those tend to be the sorts of things that cause issues when you change the design, or if your users need to zoom in on the page, etc.  So my spider-sense told me that adding a new pixel offset for centering might not be the ideal method.  :-)

I have found flex layout capable of fixing so many issues, especially centering things.  Although, it has occasionally turned out to be the wrong tool and caused issues of its own.  This happened in our subtitle rendering code not too long ago.

Reply:
Oh, I found one more button that is off-center: the "+" button on the "custom content" page.  You'll need to add the selector `.mdl-button--fab` and change `flex` to `inline-flex`.  That seems to fix both kinds of MDL buttons.

Reply:
initially there is no flex in .mdl-button--fab. when i put inline-flex in it. the result is even worst
![nnnn](https://user-images.githubusercontent.com/58463645/115449308-4dad3f80-a238-11eb-8208-fcb49b2d7e04.PNG)

i think there is problem in material-icons-round to fix both these issue either we need to put padding top by 3px or introduce line height.
![nn2](https://user-images.githubusercontent.com/58463645/115449660-b2689a00-a238-11eb-964b-f1e4ba4b8713.PNG)

Reply:
please let me know if we both have same opinion about it!

Reply:
Sorry, I should have been more clear.  You need both selectors (for --fab and --icon), and you need to set all of the following properties:

```cs
.mdl-button--fab,
.mdl-button--icon {
  /* By default, MDL icon buttons may end up vertically off-center.  This aligns
   * them to the center both vertically and horizontally. */
  display: inline-flex;
  align-items: center;
  justify-content: center;
}
```

Setting display to flex doesn't align things to the center by default.  You also need the align-items and justify-content properties, each of which aligns in a different axis.

Reply:
oh! I got it. sorry I didn't get it for the first time. 
I changed the code and everything works fine.
thanks for help.

Reply:
No worries!  Thank you for collaborating with me!